mlir/include/mlir/Tools/lsp-server-support/Transport.h

```diff
@@ -171,14 +171,44 @@ class MessageHandler {
     };
   }
 
+  /// Create an OutgoingMessage function that, when called, sends a request with
+  /// the given method and ID via the transport. Should the outgoing request be
+  /// met with a response, the response callback is invoked to handle that
+  /// response.
+  template <typename T>
+  OutgoingMessage<T> outgoingRequest(
+      llvm::StringLiteral method, llvm::json::Value id,
```

Comment:
You don't want to bind the `id` on registration, because that should generally change per invocation -> The id is the connection between the request and the response.

Reply:
Ah, thank you, good call. I've updated the patch, please take a look. Also, a little embarrassing, but because I opted to use a different function type `OutgoingRequest<T>` to handle this, if you like this version of the patch then I'll revert #90076 before landing.